docs/Reference/Principles.md

```diff
@@ -0,0 +1,78 @@
+# Technical Principles
+
+Every decision in the Fastify framework and its official plugins is guided by
+the following technical principles:
+
+1. “Zero” overhead in production;
+2. “Good” developer experience;
+3. Works great for small & big projects alike;
+4. Easy to migrate to microservices (or even serverless) and back;
+5. Security & data validation;
+6. If something could be a plugin, it likely should
+7. Easily testable;
+8. Do not monkeypatch core;
+9. Semantic versioning & Long Term Support;
+10. Specification adherence.
+
+## "Zero" Overhead in Production
+
+Fastify aims to implement its features by adding as minimal overhead to your
+application as possible.
+This is usually delivered by implementing fast algorithms and data structures,
+as well as JavaScript-specific features.
+
+Given that JavaScript does not offer zero-overhead data structures, this principle
+is at add with providing a great developer experience and providing more features,
+as usually those cost some overhead.
```

Comment:
Bit lost with this sentence as "is at add" doesn't make sense?

Reply:
```suggestion
Given that JavaScript does not offer zero-overhead data structures, this principle
is at odd with providing a great developer experience and providing more features,
as usually those cost some overhead.
```

Reply:
add -> odd